### ISSUE TITLE
IDE1006 should not trigger for P/Invoke

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn#51727]
**Version Used**: 
16.9

**Steps to Reproduce**:

```csharp
        [DllImport("mono-2.0-bdwgc.dll")]
        static extern IntPtr mono_thread_current();
```

P/Invoke requires name matching the native definition, thus unsuitable for naming analysis.

There are several variations of P/Invoke, including FCall and DllImport. The simplest way should be just ignoring all `extern` methods.
Question: how about a codebase with `EntryPoint =` set for every P/Invoke, and wants the interop method to follow their managed naming convention.

**Comment by jinujoseph** [comment]
Bringing to Design meeting to discuss "If we should change code fix to apply the entry point property" also consider alternatives like the native method class recognized by stylecop

**Comment by sharwell** [comment]
Design review conclusion: update the naming rule analyzer to not analyze or report diagnostics on items marked `extern`.
